attribute-service-api/src/main/proto/org/hypertrace/core/attribute/service/v1/attribute_metadata.proto

```diff
@@ -137,3 +138,39 @@ message AttributeMetadataFilter {
 message Empty {
 }
 
+message AttributeDefinition {
+  oneof value {
+    string span_path = 1;
+    Projection projection = 2;
+  }
+}
+
+message Projection {
+  oneof value {
+    string attribute_key = 1;
+    LiteralValue literal = 2;
+    ProjectionExpression expression = 3;
+  }
+}
+
+message LiteralValue {
+  oneof value {
+    string string_value = 1;
+    bool boolean_value = 2;
+    double float_value = 3;
+    int64 int_value = 4;
+  }
+}
+
+message ProjectionExpression {
+  ProjectionOperator operator = 1;
+  repeated Projection arguments = 2;
+}
+
+enum ProjectionOperator {
+  PROJECTION_OPERATOR_UNSET = 0;
+  PROJECTION_OPERATOR_CONCAT = 1;
+  PROJECTION_OPERATOR_HASH = 2;
+  PROJECTION_OPERATOR_DEFAULT = 3;
```

Comment:
what's the default operator?

Reply:
the idea here is something like fallback/default value. Use the value of X attribute if defined, else Y attribute. Or else literal, etc. Open to other names.